### PR TITLE
Deployment Comment `module load` Update

### LINKED ISSUE #114: Update deployment comment to reflect the removal of `.../vk83/modules/access-models`
Due to https://github.com/ACCESS-NRI/model-config-tests/pull/41#pullrequestreview-2229285965 making the module path `/g/data/vk83/modules/` instead of `/g/data/vk83/modules/access-models`, we need to update the deployment comment.

### PR DESCRIPTION
**Pull request by CodeGat**
This PR updates the deployment comment and updates the tagging logic also.

In this PR:
- **ci.yml: Updated deployment comment to have new module load, updated spack version**
- **cd.yml: Updated tagging logic similar to ACCESS-NRI/model-config-tests#55**

Closes #114

### CODE REVIEW

**Review by jo-basevi — COMMENTED** [review]
Git tag changes look good to me! Just have a question re module directory

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -227,11 +227,11 @@ jobs:
 
             This Prerelease is accessible on Gadi using:
             ```bash
-            module use /g/data/vk83/prerelease/modules/access-models/
+            module use /g/data/vk83/modules
```
[the remark this answers is not in the document]
Comment (CodeGat) [reply .github/workflows/ci.yml:230]: I'm not sure - going off https://github.com/ACCESS-NRI/build-cd/issues/114#issuecomment-2309262059 It seems like they're the same now...? @dougiesquire what exactly did you bully him into? :laughing:

[the remark this answers is not in the document]
Comment (dougiesquire) [reply .github/workflows/ci.yml:230]: Yeah they're still in a `prerelease` directory. Payu had some issue with the old module structure that includes `access-models` so @aidanheerdegen did some reshuffling that gets rid of the `access-models` directory.

**Review by jo-basevi — APPROVED** [review]
LGTM

**Pull request merged**